### PR TITLE
Allow for incremental addition of data to alt_allele [VS-52]

### PR DESCRIPTION
**Pull request by rsasch**
- changed CREATE OR REPLACE TABLE to CREATE TABLE IF NOT EXISTS for `alt_allele`
- pass around max sample id already in `alt_allele` to figure out which `vet_` table to start with and what sample IDs data to grab
- tied out by comparing `gvs-internal.rsa_gvs_quickstart_dev.alt_allele` (done with existing version) vs. `gvs-internal.rsa_gvs_quickstart_dev.alt_allele_incremental` (done in two batches with code from this branch)
- renamed WDL to GvsPopulateAltAllele

Closed https://broadworkbench.atlassian.net/browse/VS-52

### CODE REVIEW

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 1]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
 
   command <<<
     set -e
-
     echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+    # if the maximum sample_id value is evenly divisible by 4000, then max_sample_id / 4000 will
+    # give us the right vet_* table to start with; otherwise, we need to start with the next table
+    if [ $((~{max_sample_id} % 4000)) -eq 0 ]; then
+      echo $((~{max_sample_id} / 4000)) > min_vat_table_num.txt
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:118]: variablizing
```suggestion
      min_vat_table_num=$((~{max_sample_id} / 4000))
```

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 2]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
...
+    else
+      echo $(((~{max_sample_id} / 4000) + 1)) > min_vat_table_num.txt
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:120]:
```suggestion
      min_vat_table_num=$(((~{max_sample_id} / 4000) + 1))
```

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 3]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
...
+    fi
+
+    # use the number calculated from the above math to get the vet_* table names grab data from
     bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false ~{bq_labels} \
-    'SELECT table_name FROM `~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES` WHERE table_name LIKE "vet_%" ORDER BY table_name' > vet_tables.csv
+    "SELECT table_name FROM \`~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES\` WHERE table_name LIKE 'vet_%' AND CAST(SUBSTRING(table_name, 5) AS INT64) >= $(cat min_vat_table_num.txt)" > vet_tables.csv
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:125]:
```suggestion
    "SELECT table_name FROM \`~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES\` WHERE table_name LIKE 'vet_%' AND CAST(SUBSTRING(table_name, length('vet_') + 1) AS INT64) >= ${min_vat_table_num}" > vet_tables.csv
```

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 4]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
...
+
+    # if the maximum sample_id value is evenly divisible by 4000, then max_sample_id / 4000 will
+    # give us the right vet_* table to start with; otherwise, we need to start with the next table
+    if [ $((~{max_sample_id} % 4000)) -eq 0 ]; then
+      echo $((~{max_sample_id} / 4000)) > min_vat_table_num.txt
+    else
+      echo $(((~{max_sample_id} / 4000) + 1)) > min_vat_table_num.txt
+    fi
+
+    # use the number calculated from the above math to get the vet_* table names grab data from
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:123]:
```suggestion
    # use the number calculated from the above math to get the vet_* table names to grab data from
```

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 5]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
...
     bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false ~{bq_labels} \
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:124]:
```suggestion
    bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false ~{bq_labels} --print_header=false \
```

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 6]
```diff
@@ -64,10 +110,19 @@ task GetVetTableNames {
...
-    'SELECT table_name FROM `~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES` WHERE table_name LIKE "vet_%" ORDER BY table_name' > vet_tables.csv
+    "SELECT table_name FROM \`~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES\` WHERE table_name LIKE 'vet_%' AND CAST(SUBSTRING(table_name, 5) AS INT64) >= $(cat min_vat_table_num.txt)" > vet_tables.csv
 
     # remove the header row from the CSV file
     sed -i 1d vet_tables.csv
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsCreateAltAllele.wdl:128]: if the header is suppressed above
```suggestion
```

**Review by mcovarr — COMMENTED** [review]
some draft nitpicks, looks promising!

File: scripts/variantstore/wdl/GvsCreateAltAllele.wdl [hunk 7]
```diff
@@ -49,10 +58,47 @@ workflow GvsCreateAltAllele {
...
+  meta {
+    # because this is being used to determine the current state of the GVS database, never use call cache
+    volatile: true
+  }
+
+  command <<<
+    set -e
+
+    echo "project_id = ~{project_id}" > ~/.bigqueryrc
+    bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false \
```
[the remark this answers is not in the document]
Comment (rsasch) [reply scripts/variantstore/wdl/GvsCreateAltAllele.wdl]: Looks like the `print_header` flag is only for `bq extract` and doesn't exist for `bq query`: https://cloud.google.com/bigquery/docs/reference/bq-cli-reference#bq_query.

[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/wdl/GvsCreateAltAllele.wdl:76]: Bummer. This could pipe to `sed 1d` for the same effect though.

**Review by mcovarr — APPROVED** [review]

File: scripts/variantstore/wdl/GvsPopulateAltAllele.wdl [hunk 8]
```diff
@@ -166,10 +221,11 @@ task PopulateAltAlleleTable {
       --call_set_identifier ~{call_set_identifier} \
       --query_project ~{project_id} \
       --vet_table_name ~{vet_table_name} \
-      --fq_dataset ~{project_id}.~{dataset_name}
+      --fq_dataset ~{project_id}.~{dataset_name} \
+      --max_sample_id ~{max_sample_id} \
   >>>
   runtime {
-    docker: "us.gcr.io/broad-dsde-methods/variantstore:ah_var_store_2022_08_16"
+    docker: "us.gcr.io/broad-dsde-methods/variantstore:rsa_vs_52_incremental_alt_allele_2022_08_17"
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsPopulateAltAllele.wdl:228]: Once ready to merge, can your create and use 'ah_var_store_2022_08_22' (or whatever date). And update everywhere?

File: scripts/variantstore/wdl/extract/populate_alt_allele_table.py [hunk 9]
```diff
@@ -39,6 +39,7 @@ def populate_alt_allele_table(call_set_identifier, query_project, vet_table_name
     parser.add_argument('--query_project',type=str, help='Google project where query should be executed', required=True)
     parser.add_argument('--vet_table_name',type=str, help='vet table name to ingest', required=True)
     parser.add_argument('--fq_dataset',type=str, help='project and dataset for data', required=True)
+    parser.add_argument('--max_sample_id',type=str, help='Maximum value of sample_id already loaded', required=True)
```
Comment (gbggrant) [inline scripts/variantstore/wdl/extract/populate_alt_allele_table.py:42]: Annoying nit:
```suggestion
    parser.add_argument('--max_sample_id', type=str, help='Maximum value of sample_id already loaded', required=True)
```

File: scripts/variantstore/wdl/GvsPopulateAltAllele.wdl [hunk 10]
```diff
@@ -64,16 +109,25 @@ task GetVetTableNames {
...
     bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false ~{bq_labels} \
-    'SELECT table_name FROM `~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES` WHERE table_name LIKE "vet_%" ORDER BY table_name' > vet_tables.csv
+    "SELECT table_name FROM \`~{project_id}.~{dataset_name}.INFORMATION_SCHEMA.TABLES\` WHERE table_name LIKE 'vet_%' AND CAST(SUBSTRING(table_name, length('vet_') + 1) AS INT64) >= ${min_vat_table_num}" > vet_tables.csv
 
     # remove the header row from the CSV file
     sed -i 1d vet_tables.csv
   >>>
   runtime {
-    docker: "gcr.io/google.com/cloudsdktool/cloud-sdk:305.0.0"
+    docker: "gcr.io/google.com/cloudsdktool/cloud-sdk:398.0.0"
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsPopulateAltAllele.wdl:130]: Can we update to 398 everywhere?

**Review by gbggrant — APPROVED** [review]
LGTM

File: scripts/variantstore/wdl/GvsPopulateAltAllele.wdl [hunk 11]
```diff
@@ -103,7 +157,7 @@ task CreateAltAlleleTable {
 
     echo "project_id = ~{project_id}" > ~/.bigqueryrc
     bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false ~{bq_labels} \
-    'CREATE OR REPLACE TABLE `~{project_id}.~{dataset_name}.alt_allele` (
+    'CREATE TABLE IF NOT EXISTS `~{project_id}.~{dataset_name}.alt_allele` (
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsPopulateAltAllele.wdl:160]: ooh, I wasn't aware of this command… nice!

File: scripts/variantstore/wdl/extract/populate_alt_allele_table.py [hunk 12]
```diff
@@ -23,9 +23,9 @@ def populate_alt_allele_table(call_set_identifier, query_project, vet_table_name
     alt_allele_positions = Path('alt_allele_positions.sql').read_text()
     fq_vet_table = f"{fq_dataset}.{vet_table_name}"
     query_with = f"""INSERT INTO `{fq_dataset}.alt_allele`
-                WITH 
-                  position1 as (select * from `{fq_vet_table}` WHERE call_GT IN ('0/1', '1/0', '1/1', '0|1', '1|0', '1|1', '0/2', '0|2','2/0', '2|0')),
-                  position2 as (select * from `{fq_vet_table}` WHERE call_GT IN ('1/2', '1|2', '2/1', '2|1'))"""
+                WITH
+                  position1 as (select * from `{fq_vet_table}` WHERE call_GT IN ('0/1', '1/0', '1/1', '0|1', '1|0', '1|1', '0/2', '0|2','2/0', '2|0') AND sample_id > {max_sample_id}),
```
Comment (kcibul) [inline scripts/variantstore/wdl/extract/populate_alt_allele_table.py:27]: smart, I like the use of `>` rather than some `IN` kind of clause

**Pull request merged**